### PR TITLE
Read README as utf-8 text and set the content_type to text/markdown

### LINKED ISSUE #17: README on pypi.org not readable
https://pypi.org/project/bump2version/ shows content of README, which is broken.

there seem to be two reasons:
- the value for `long_description` is provided as bytes and not as a string
- the content if formatted as markdown and not reStructuredText

With refactored https://pypi.org well readable README can serve as great presentation of the package and `bump2version` can definitely do better.

If reStructuredText seems too difficult, I volunteer to write it properly. It is not so difficult and such great tool as bump2version deserves good presentation.

### PR DESCRIPTION
**Pull request by florisla**
This is an improvement on #24 and fixes #17.

The generated wheel file contains the correct content_type and 'README' content.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-  r"\1",
-  str(open('README.md', 'rb').read()).replace(description, '')
-)
+# Import the README and use it as the long-description.
+# Note: this will only work if 'README.md' is present in your MANIFEST.in file!
+here = os.path.abspath(os.path.dirname(__file__))
+try:
+    with open(os.path.join(here, 'README.md'), encoding='utf-8') as f:
+        long_description = '\n' + f.read()
+except FileNotFoundError:
```
Comment (ekohl) [inline setup.py]: This exception was introduced in Python 3.something. Sadly it doesn't work on py2.

File: setup.py [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
-  r"\1",
-  str(open('README.md', 'rb').read()).replace(description, '')
-)
+# Import the README and use it as the long-description.
+# Note: this will only work if 'README.md' is present in your MANIFEST.in file!
+here = os.path.abspath(os.path.dirname(__file__))
+try:
+    with open(os.path.join(here, 'README.md'), encoding='utf-8') as f:
+        long_description = '\n' + f.read()
+except IOError:
```
Comment (ekohl) [inline setup.py]: I'm actually not sure we should be catching that exception. It should always exist and otherwise it's an error.

File: setup.py [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
-  "\`(.*)\<#.*\>\`\_",
-  r"\1",
-  str(open('README.md', 'rb').read()).replace(description, '')
-)
+# Import the README and use it as the long-description.
+# Note: this will only work if 'README.md' is present in your MANIFEST.in file!
+here = os.path.abspath(os.path.dirname(__file__))
+try:
+    with open(os.path.join(here, 'README.md'), encoding='utf-8') as f:
+        long_description = '\n' + f.read()
```
Comment (ekohl) [inline setup.py]: Why start with a newline?

Comment (florisla) [reply setup.py]: It just looks nicer in the `*.dist-info/METADATA` file in the wheel package.

File: setup.py (hunk 2 above)
Comment (florisla) [reply setup.py]: OK, will update.

**Comment by ekohl** [comment]
Could you rebase on master and see if it fixes the tests on travis with 3.4?

**Comment by florisla** [comment]
No luck with AppVeyor or Travis... any suggestions?

**Comment by c4urself** [comment]
@florisla -- there's work on the AppVeyor passing tests in this PR: https://github.com/c4urself/bump2version/pull/33 -- don't strictly need that passing for a merge here.

**Pull request merged**